runcontainer.go

```diff
@@ -2281,7 +2281,7 @@ func runArrayUnionToRuns(rc *runContainer16, ac *arrayContainer) ([]interval16,
 			pos2++
 		}
 	}
-	cardMinusOne += previousInterval.length + 1
+	cardMinusOne += previousInterval.length
```

Comment:
is it worth renaming `length` to `lengthMinusOne`? I think that would have made this bug a lot more obvious.